services/web/server/src/simcore_service_webserver/resource_usage/_service_runs_handlers.py

```diff
@@ -135,3 +143,25 @@ async def list_resource_usage_services(request: web.Request):
         text=page.json(**RESPONSE_MODEL_POLICY),
         content_type=MIMETYPE_APPLICATION_JSON,
     )
+
+
+@routes.post(
+    f"/{VTAG}/services/-/resource-usages:export", name="export_resource_usage_services"
+)
+@login_required
+@permission_required("resource-usage.read")
+@_handle_resource_usage_exceptions
+async def export_resource_usage_services(request: web.Request):
+    req_ctx = _RequestContext.parse_obj(request)
+    query_params = parse_request_query_parameters_as(
+        _ListServicesResourceUsagesQueryParams, request
+    )
+    download_url = await api.export_usage_services(
+        app=request.app,
+        user_id=req_ctx.user_id,
+        product_name=req_ctx.product_name,
+        wallet_id=query_params.wallet_id,
+        order_by=parse_obj_as(OrderBy | None, query_params.order_by),
+        filters=parse_obj_as(ServiceResourceUsagesFilters | None, query_params.filters),
+    )
+    raise web.HTTPFound(location=f"{download_url}")
```

Comment:
So this actually waits for the report to be generated. I have an observation.

With load (when you have lots of elects to put into the report) this will generate a long running HTTP request. There is no feedback on how the report generation progress is going and if the request got stuck.

I would at least have this run via long running tasks (there is an aiohttp version of it) and I think the frontend already has functionality for it. Also they have functionality for transmitting back some progress of this operation.

So finally the progress is what I think it's really lacking here. As a user I would like to know how much time it will take for my very long report to finish generating.

Reply:
The same as above and discussed in person